models/Ion_auth_model.php

```diff
@@ -360,52 +373,55 @@ public function rehash_password_if_needed($hash, $identity, $password)
 	}
 
 	/**
-	 * Validates and removes activation code.
+	 * Check a user activation code
 	 *
-	 * @param int|string $id
-	 * @param bool       $code		if omitted, simply activate the user without check
+	 * @param int|string $id		the user identifier
+	 * @param bool       $code		the activation code 
+	 * 								It's the *user* one, containing "selector.validator"
+	 * 								the one you got in activation_code member
+	 * 								if omitted, simply activate the user without check
 	 *
 	 * @return bool
 	 * @author Mathew
 	 */
-	public function activate($id, $code = FALSE)
+	public function check_user_code_activation($id, $code)
 	{
-		$this->trigger_events('pre_activate');
-
 		$token = $this->_retrieve_selector_validator_couple($code);
+	
+		$query = $this->db->select([$this->identity_column, 'activation_code'])
+							->where('activation_selector', $token->selector)
+							->where('id', $id)
```

Comment:
This function might be even more useful if it were to return the user object, but maybe that's a bridge too far?

If not, I could just use the $uid because it's been validated with the activation key in this function.

I'm not good with backwards compatibility, I'd break the entire world :-)

Reply:
I understand.
You would have preferred the same type of function as `get_user_by_forgotten_password_code()`?

@benedmunds what do you think about modifying `check_user_code_activation()` to `get_user_by_activation_code()`?

Reply:
I really think it's fine as is but I'm not against the change.  Or maybe add another method for get_user_by_activation_code that check_user_code_activation uses.

Reply:
I didn't want to make it too much complicated, so I simply renamed it `get_user_by_activation_code()` and change its behavior to return the user on success. The id is no longer needed to get the user, but is still later checked in `activate()`.

The nice thing is that this new function is almost the same as `get_user_by_forgotten_password_code()`, which makes sense.

Reply:
This looks good to me 👍